mongo/pagination/pagination.go

```diff
@@ -24,11 +25,14 @@ type Meta struct {
 	perPageNum  int
 }
 
+type QueryFunc func(ctx context.Context, collectionName string, query bson.M, sortFields ...document.SortField) (*mgo.Query, document.Closer)
+
 type PaginateOpts struct {
 	PageNumber  int
 	AmountItems int
 	Query       bson.M
 	SortOrder   string
+	QueryFunc  QueryFunc
```

Comment:
Suggestion: add a comment to say what this property does. I need to read the code to find out that it replaces the default WhereQuery. It would also be good to know if it respects (or ignores) the "PageNumber", "AmountItems" and "SortOrder" properties.

Reply:
Ack